evalml/pipelines/components/component_base_meta.py

```diff
@@ -19,6 +19,8 @@ def _check_for_fit(self, X=None, y=None):
             klass = type(self).__name__
             if not self._is_fitted and self.needs_fitting:
                 raise ComponentNotYetFittedError(f'This {klass} is not fitted yet. You must fit {klass} before calling {method.__name__}.')
+            elif method.__name__ == 'inverse_transform':
```

Comment:
This is a short-term fix to make the component metaclass aware of the `inverse_transform` method. The current implementation will not work because if `y` is `None`, the method will run with only the `X` argument which is not what we want.

I think we can change this implementation to something like:
```python
def _check_for_fit(self, *args, **kwargs):
    klass = type(self).__name__
    if not self._is_fitted and self.needs_fitting:
        raise ComponentNotYetFittedError(f'This {klass} is not fitted yet. You must fit {klass} before calling {method.__name__}.')
    return method(self, *args, **kwargs)
```

I think this would be more maintainable in the long-term.

Reply:
@ParthivNaresh I think we're currently raising an exception in `inverse_transform` if `y` is `None`.  The problem is that the meta class doesn't pass the `y` argument if it's None to `inverse_transform` in the current implementation. This causes an error because we're expecting a `y` argument, even if it's None.